modules/calib3d/src/calibinit.cpp

```diff
@@ -1937,6 +1938,7 @@ bool ChessBoardDetector::processQuads(std::vector<cv::Point2f>& out_corners, int
         if (count > 0 || (-count > (int)out_corners.size()))
         {
             // copy corners to output array
+            out_corners.clear();
```

Comment:
There is `out_corners.resize(0);` at the function beginning (line 1886). M.b. it should be replaced with clear() call and the condition may be reduced.

Reply:
Right, there is one line 1886 but what if you enter this block (1938 to 1951), fill out_corners and actually fail the condition like 1946?
You then can push_back again in the next loop iteration (from loop at 1899) while `out_corners` has not been cleared.